### PR TITLE
Aligned eq maxwell

### PR DESCRIPTION
**Pull request by DorisReiter**
Updated Maxwell's equations and added boundary conditions.

### CODE REVIEW

File: docs/math/maxwell.md [hunk 1]
```diff
@@ -1,70 +1,147 @@
...
-    =
-    \varepsilon\frac{\partial \vec{\mathcal{E}}}{\partial t}
-$$ (maxwell_no_sources)
+Now putting these back in the Maxwell equations we obtain 
+
+$$\begin{aligned}
+    \mathbf{\nabla}\cdot\mathbf{D}\left(\mathbf{r},t\right)=& \mathbf{\nabla}\cdot\left(\varepsilon_0 \mathbf{E}\left(\mathbf{r},t\right) + \mathbf{P}\left(\mathbf{r},t\right) \right) \\
+        \Rightarrow 
+        \mathbf{\nabla}\cdot\mathbf{E}\left(\mathbf{r},t\right) =& \frac{1}{\varepsilon_0} \left(  \mathbf{\nabla}\cdot\mathbf{D}\left(\mathbf{r},t\right) -  \mathbf{\nabla}\cdot\mathbf{P}\left(\mathbf{r},t\right) \right)
+        = \frac{1}{\varepsilon_0} \left( \rho_{\text{ext}}\left(\mathbf{r},t\right)+\rho_{\text{int}}\left(\mathbf{r},t\right) \right)\varrho\,
```
Comment (HelgeGehring) [inline docs/math/maxwell.md:57]:
```suggestion
        = \frac{1}{\varepsilon_0} \left( \rho_{\text{ext}}\left(\mathbf{r},t\right)+\rho_{\text{int}}\left(\mathbf{r},t\right) \right)\,
```

The $\varrho$ shouldn't be there, right?

File: docs/math/maxwell.md [hunk 2]
```diff
@@ -1,70 +1,147 @@
...
+
+$$\begin{aligned}
+    \mathbf{\nabla}\cdot \left( \varepsilon\left(\mathbf{r}\right)  \mathbf{E}\left(\mathbf{r},t\right) \right) &=& \varrho \left(\mathbf{r},t\right)\\
+     \mathbf{\nabla}\cdot \left( \mu \left(\mathbf{r}\right) \mathbf{H}\left(\mathbf{r},t\right) \right) &=& 0 \\
+    \mathbf{\nabla}\times\mathbf{E}\left(\mathbf{r},t\right) &=& - \mu \frac{\partial \mathbf{H}\left(\mathbf{r},t\right)} {\partial t} \\ 
+    \mathbf{\nabla}\times\mathbf{H}\left(\mathbf{r},t\right) &=& \mathbf{J}_{\text{ext}}\left(\mathbf{r},t\right) + \varepsilon \frac{\partial \mathbf{E}\left(\mathbf{r},t\right)} {\partial t} \,.
+\end{aligned}$$
+
+### Piecewise constant materials and boundary conditions
+It is instructive to condsider the well-known case of an interface $I$ between two dielectric materials, which appear in many devices. We assume an interface between two materials called $1$ with dielectric constant $\varepsilon_1$ and $2$ with dielectric constant $\varepsilon_2$. The surface is defined by the normal vector of the interface $\mathbf{n}_{I}$ and there are no external surface charges or currents. For simplicity, we surpress the index  $\left(\mathbf{r},t\right)$ here.
```
Comment (HelgeGehring) [inline docs/math/maxwell.md:100]:
```suggestion
It is instructive to condsider the well-known case of an interface $I$ between two dielectric materials, which appear in many devices. We assume an interface between two materials called $1$ with dielectric constant $\varepsilon_1$ and $2$ with dielectric constant $\varepsilon_2$. The surface is defined by the normal vector of the interface $\mathbf{n}_{I}$ and there are no external surface charges or currents. For simplicity, we surpress the arguments $\left(\mathbf{r},t\right)$ here. 
```

File: docs/math/maxwell.md [hunk 3]
```diff
@@ -1,70 +1,147 @@
...
+
+$$\begin{aligned}
+    \varepsilon\left(\mathbf{r}\right) = \varepsilon_0 \varepsilon_r\left(\mathbf{r}\right)\\
+    \mu\left(\mathbf{r}\right) = \mu_0 \mu_r\left(\mathbf{r}\right)
+\end{aligned}$$ 
+
+We remind that it holds that 
+
+$$\begin{aligned}
+    c^2=\frac{1}{\mu_0\varepsilon_0}
```
Comment (HelgeGehring) [inline docs/math/maxwell.md:87]: should we call it $c_0$ instead of $c$? 
```suggestion
    c_0^2=\frac{1}{\mu_0\varepsilon_0}
```

**Review by HelgeGehring — CHANGES_REQUESTED** [review]
Looks great to me!

Just two little things which I found (see above), could you have a look if I'm right?

Besides that, people start asking about anisotropic permittivities.
We can probably include them directly by just writing $\varepsilon$ and $\mu$ as $\boldsymbol{\varepsilon}$ and $\boldsymbol{\mu}$, right?
(seems like `mathbf` doesn't work for them but `boldsymbol` works fine.

File: docs/math/maxwell.md (hunk 1 above)
Comment (DorisReiter) [reply docs/math/maxwell.md:57]: true and the comma shouldn't be there, too

File: docs/math/maxwell.md (hunk 3 above)
Comment (DorisReiter) [reply docs/math/maxwell.md:87]: I think we should stick with "c" as speed of light

**Comment by DorisReiter** [comment]
Corrected another typo and changes arguments -> dependencies

**Comment by DorisReiter** [comment]
> 

If we consider anisotropic material, we need to change eps and mu into tensors, not vectors. I think it will not help here, but could be a side comment for future references. 
btw. for plasmonic problems (metals), we should also not drop the time dependence, so the most general case is keeping everything, though this might not be the most instructive case. 
My suggestion would be to leave it for now and extend it later.

**Review by HelgeGehring — APPROVED** [review]

**Comment by HelgeGehring** [comment]
Sounds great! I just closed/reopened the PR to run the automatic tests, afterwards I'll merge :)

**Pull request merged**